metron-platform/metron-writer/src/main/java/org/apache/metron/writer/hdfs/SourceHandlerCallback.java

```diff
@@ -29,7 +34,10 @@
   }
 
   public void removeKey() {
-    sourceHandlerMap.remove(key);
+    SourceHandler removed = sourceHandlerMap.remove(key);
+    removed.close(); // If it's getting removed, we want to close it to ensure things like Timers are ended.
+    LOG.debug("Removed {} -> {}", key, removed);
+    LOG.debug("Current state of sourceHandlerMap: {}", sourceHandlerMap);
```

Comment:
Would it make sense to just push this into a single log statement?  I'm worried that out-of-order logging would make it difficult to connect the dots that after I removed X, the current state is Y.  If its one statement, you don't have that potential problem.